### ISSUE TITLE
Icon widget broken by 4.0.6

### ISSUE THREAD

**Issue by lancasterJ** [AppDaemon/appdaemon#1234]
Icon widget stopped displaying the icon and state text when upgraded to 4.0.6.  Before updating to 4.0.6 this code

```
garbage:
    widget_type: icon
    entity: sensor.garbage
    title: Garbage
    title2: collection
    state_text: 1
    icons:
      "today":
        icon: mdi-delete-circle
        style: "color: blue"
      "tomorrow":
        icon: mdi-delete-restore
        style: "color: black"
      "default":
        icon: mdi-delete-empty
        style: "color: white"
```

produced 

![image](https://user-images.githubusercontent.com/63460573/112718576-dfe94e80-8eeb-11eb-94ef-e7e54bda85d7.png)

After the upgrade the following is displayed

![image](https://user-images.githubusercontent.com/63460573/112718594-01e2d100-8eec-11eb-9cc0-2faf5e2e1cd9.png)

Commenting out the "default" section results in the state text being displayed, but not the icon.

```
garbage:
    widget_type: icon
    entity: sensor.garbage
    title: Garbage
    title2: collection
    state_text: 1
    icons:
      "today":
        icon: mdi-delete-circle
        style: "color: blue"
      "tomorrow":
        icon: mdi-delete-restore
        style: "color: black"
      # "default":
        # icon: mdi-delete-empty
        # style: "color: white"
```

![image](https://user-images.githubusercontent.com/63460573/112718645-4b332080-8eec-11eb-9ff6-891c85ad46af.png)

**Comment by Odianosen25** [comment]
@lancasterJ,

Please can you advise on what kind of device and browser you are using?

Regards

**Comment by lancasterJ** [comment]
@Odianosen25,

The browsers:
- Firefox 86.0.1 (64-bit) for Windows.
- Chromium 78.0.3904.108 Built on Raspbian , running on Raspbian 10.
- Safari for iOS 14.4.

The App Daemon service is running on a Raspberry Pi 3 using the AppDaemon 4 (0.5.0) add-on for Home Assistant 2021.3.4.

Home Assistant details:
- core-2021.3.4
- Python Version 3.8.7
- Operating System Version 5.4.83-v7
- CPU Architecture armv7l
- Host Operating System | Home Assistant OS 5.12
- Supervisor Version | supervisor-2021.03.6
- Docker Version | 19.03.13

Regards
John

**Comment by ReneTode** [comment]
icon widget is broken in some browsers on some systems.
this will be repaired in the next release.

**Comment by lancasterJ** [comment]
@Odianosen25 and @ReneTode,

I am sorry to report that 4.0.7 did not resolve this issue.  

I have install 4.0.7, restarted AppDaemon, cleared browser caches.  The issue still exists on the three browsers listed above. 

I have also tested using Microsoft Edge Version 89.0.774.63 (Official build) (64-bit).  The issue also exists when this browser is used.

Regards
John

**Comment by ReneTode** [comment]
i just tested myself and indeed there is a poblem.
when the state isnt in the icon list it shows up as 1.
but the icons show up.

normally i always said: use chrome if you can, because we dont support every browser with dashboard.
we try to, but only garantee chrome, because thats the only browser i use.

im not a dev anymore and i dont know how andrew would react to it now.

but from what i see there are errors in the widget anyway, so ill see what i can do.

in the meantime you can always get the old widget back.
just download it from here:
https://github.com/AppDaemon/appdaemon/tree/4.0.5/appdaemon/widgets

all you need is the baseicon dir, and you need to copy it to you appdaemon config area
to the dir
<your ad config>/custom_widgets/baseicon

**Comment by ReneTode** [comment]
found the error, made a PR for it. will be fixed in the next release.
(after that it will probably work again for all browsers.)

when the PR is merged you can try it by downloading the files from 
https://github.com/AppDaemon/appdaemon/tree/dev/appdaemon/widgets

and then place them in the custom_widgets directory.

**Comment by Odianosen25** [comment]
@lancasterJ,

This is not fixed in dev thanks to @ReneTode, kindly confirm when you can so we can close this and do a quick release.

Kind regards

**Comment by ReneTode** [comment]
autocorrect made not form now :P

**Comment by Odianosen25** [comment]
Lols just noticed thanks 😊

**Comment by Odianosen25** [comment]
This has been fixed in AD 4.0.8
